src/edges.jl

```diff
@@ -0,0 +1,59 @@
+# Edges utilities
+# Edge stores
+# - an id corresponding to index in graph
+# - a noise, or a collection of noise (for stagewise noise)
+# - an optimization problem (implemented with JuMP or MOI)
+
+abstract type AbstractEdge end
```

Comment:
Isn't already defined as LightGraphs.AbstractEdge ? We need to decide whether we want to implement LightGraphs interface or create a new one.
In LightGraph, an edge is just `Edge(src, dst)`. In this case, we need to do e.g.
```julia
sync!(sp::AbstractStochasticProgram, edge)
```
instead of
```julia
sync!(edge)
```
because the edge only contains the incoming and outgoing edge and the rest is stored in the stochastic program.

Reply:
Yes presumably all these methods should take `sp` as the first argument.

Reply:
ok, it seems that I do not have the same vision as yours :p But you are right concerning `sp`. 

Concerning Lightgraphs, the arguments would be:

**Pros **
- we would be able to use already existing methods in a lightweight package
- able also to plot easily our graphs! 

**Cons** 
- We have to find a proper (and efficient) way to overwrap Lightgraphs (for instance, how to loop easily on a node's outgoing edges?)

What do you think?

Reply:
I might miss some of the argument, but why not use MetaGraphs ? As far as I know it's a very light package building on LightGraph allowing to attach to an edge or nodes any object.

Reply:
>We have to find a proper (and efficient) way to overwrap Lightgraphs (for instance, how to loop easily on a node's outgoing edges?)

The existing functionality `out_neighbors` is not efficient?

Reply:
`out_neighbors` allows to loop upon outgoing nodes, and not outgoing edges. I do not find a proper function to loop upon outgoing edges, but I am not an expert on `Lightgraphs` ...

Reply:
@frapac The function `out_edges` has been removed, I think we now need to do `Edge.(u, out_neighbors(g, u))` instead of `out_edges(g, u)`.
@leclere I initially wanted to use MetaGraphs in StructDualDynProg but when you look at it [here](https://github.com/JuliaGraphs/MetaGraphs.jl/blob/master/src/metadigraph.jl#L2-L7) and [here](https://github.com/JuliaGraphs/MetaGraphs.jl/blob/master/src/MetaGraphs.jl#L44), it seems inefficient because all properties are stored in a `Dict{Symbol, Any}`. MetaGraphs cannot really do better at his abstract level (or maybe using macros as we do in MathOptInterfaceUtilities :-P) but we can do faster by storing dictionnaries like `proba::Dict{SimpleEdge{Int}, Float64}` which will be typed.